addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -197,6 +197,48 @@ export class PosStore extends Reactive {
         this.computeProductPricelistCache();
     }
 
+    async deleteOrders(orders, serverIds = []) {
+        const ids = new Set();
+        for (const order of orders) {
+            if (order && (await this._onBeforeDeleteOrder(order))) {
+                if (Object.keys(order.last_order_preparation_change).length > 0) {
+                    await this.sendOrderInPreparationUpdateLastChange(order, true);
+                }
+
+                const cancelled = this.removeOrder(order, true);
+                if (!cancelled) {
+                    return false;
+                } else if (typeof order.id === "number") {
+                    ids.add(order.id);
+                }
+            }
+        }
+
+        if (serverIds.length > 0) {
+            for (const id of serverIds) {
+                if (typeof id !== "number") {
+                    continue;
+                }
+                ids.add(id);
+            }
+        }
+
+        if (ids.size > 0) {
+            this.pendingOrder.delete.clear();
+            await this.data.call("pos.order", "action_pos_order_cancel", [Array.from(ids)]);
+        }
+
+        return true;
+    }
```

Comment:
I notice that this is a moved code. However, the handling of `this.pendingOrder.delete` is different now from the original. Before, it became `idToCancel`, now, it's only cleared.

Reply:
This is no longer necessary, the information is in the ids set.